src/__tests__/stream.test.ts

```diff
@@ -0,0 +1,148 @@
+
+  let clientConn: IlpStream.Connection
+  let clientStream: IlpStream.DataAndMoneyStream
+
+  const port = await (getPort() as Promise<number>)
+
+  const clientPlugin = new EthereumPlugin({
+    role: 'client',
+    ethereumPrivateKey: process.env.PRIVATE_KEY_A,
+    ethereumProvider: process.env.ETHEREUM_PROVIDER,
+    // @ts-ignore
+    server: `btp+ws://userA:secretA@localhost:${port}`,
+    outgoingChannelAmount: convert('0.01', Unit.Eth, Unit.Gwei),
+    balance: {
+      settleTo: SENDER_SETTLE_TO,
+      settleThreshold: convert('0.0019', Unit.Eth, Unit.Gwei)
+    }
+  })
+
+  const serverPlugin = new EthereumPlugin({
+    role: 'server',
+    ethereumPrivateKey: process.env.PRIVATE_KEY_B,
+    ethereumProvider: process.env.ETHEREUM_PROVIDER,
+    // @ts-ignore
+    debugHostIldcpInfo: {
+      assetCode: 'ETH',
+      assetScale: 9,
+      clientAddress: 'private.ethereum'
+    },
+    port,
+    maxPacketAmount: convert('0.0001', Unit.Eth, Unit.Gwei),
+    outgoingChannelAmount: convert('0.01', Unit.Eth, Unit.Gwei),
+    incomingChannelFee: INCOMING_FEE,
+    balance: {
+      maximum: RECEIVER_MAX_BALANCE
+    }
+  })
+
+  clientPlugin.registerMoneyHandler(Promise.resolve)
+  serverPlugin.registerMoneyHandler((amount: string) => {
+    actualReceived = actualReceived.plus(amount)
+  })
+
+  await serverPlugin.connect()
+  await clientPlugin.connect()
+
+  // Setup the receiver (Ethereum server, Stream server)
+  const streamServer = await IlpStream.createServer({
+    plugin: serverPlugin,
+    receiveOnly: true
+  })
+
+  await new Promise(async resolve => {
+    streamServer.once('connection', (conn: IlpStream.Connection) => {
+      serverConn = conn
+
+      serverConn.once('stream', (stream: IlpStream.DataAndMoneyStream) => {
+        stream.setReceiveMax(Infinity)
+        serverStream = stream
+
+        resolve()
+      })
+    })
+
+    // Setup the sender (Ethereum client, Stream client)
+    clientConn = await IlpStream.createConnection({
+      plugin: clientPlugin,
+      ...(streamServer.generateAddressAndSecret())
+    })
+
+    clientStream = clientConn.createStream()
+    clientStream.setSendMax(AMOUNT_TO_SEND)
+  })
+
+  serverStream!.on('money', () => {
+    const amountPrefunded = actualReceived.minus(serverConn.totalReceived)
+
+    t.true(amountPrefunded.gte(RECEIVER_MAX_BALANCE), 'amount prefunded to server is always at least the max balance')
+    t.true(amountPrefunded.lte(SENDER_SETTLE_TO), 'amount prefunded to server is never greater than settleTo amount')
+  })
+
+  await t.notThrowsAsync(serverStream!.receiveTotal(AMOUNT_TO_SEND), 'client streamed the total amount of packets to the server')
+
+  // Wait 10 seconds for sender to finish sending the remaining payment channel claims
+  await new Promise(resolve => {
+    setTimeout(resolve, 10000)
+  })
+
+  t.true(actualReceived.gte(AMOUNT_TO_SEND.minus(INCOMING_FEE)), 'server received at least as much money as the client sent')
```

Comment:
Would this fail if the network fees increase?

Reply:
That test should still work. `INCOMING_FEE` is a fixed initiation fee that the connector charges when it gets a claim from a new channel (to cover the cost of them claiming it, although right now it's not dynamic to the cost of the claim tx).

However, the tests _would_ fail if the fee to open a channel was greater than the `settleTo` amount in the sender's balance configuration (since even on the client, the fee to open a channel is added to/debited from the balance). That said, the prefund amount is `0.002` eth, so the transaction fees should almost always be less than that.